releasenotes/notes/hostname-detection-container-e12cf0608270c9d4.yaml

```diff
@@ -0,0 +1,6 @@
+---
+upgrade:
+  - |
+    Starting Agent 7.40, the Agent will fail to start when unable to determine hostname instead of silently using unrelevant hostname (usually, a container id).
```

Comment:
```suggestion
    Starting with Agent 7.40, the Agent will fail to start when it is unable to determine the hostname, instead of silently using an irrelevant hostname (usually, a container ID).
```